plugins/plant.cpp

```diff
@@ -0,0 +1,627 @@
+// Grow and remove shrubs or trees.
+
+#include <iostream>
+#include <iomanip>
+#include <map>
+#include <algorithm>
+#include <vector>
+#include <string>
+
+#include "Debug.h"
+#include "Core.h"
+#include "Console.h"
+#include "Export.h"
+#include "LuaTools.h"
+#include "PluginManager.h"
+#include "TileTypes.h"
+
+#include "modules/Gui.h"
+#include "modules/MapCache.h"
+#include "modules/Maps.h"
+
+#include "df/block_square_event_grassst.h"
+#include "df/map_block.h"
+#include "df/map_block_column.h"
+#include "df/plant.h"
+#include "df/plant_raw.h"
+#include "df/world.h"
+
+using std::vector;
+using std::string;
+using namespace DFHack;
+
+DFHACK_PLUGIN("plant");
+REQUIRE_GLOBAL(world);
+
+namespace DFHack
+{
+    DBG_DECLARE(plant, log, DebugCategory::LINFO);
+}
+
+struct cuboid
+{
+    int16_t x_min = -1;
+    int16_t x_max = -1;
+    int16_t y_min = -1;
+    int16_t y_max = -1;
+    int16_t z_min = -1;
+    int16_t z_max = -1;
+
+    bool isValid() const
+    {   // False if any bound is < 0
```

Comment:
shouldn't it also return false if any max is less than the relevant min?

Reply:
Probably, since I didn't define any setter functions to prevent that.